homeassistant/components/xiaomi_miio/air_quality.py

```diff
@@ -0,0 +1,193 @@
+                self._model = info.model
+                self._mac_address = info.mac_address
+                self._sw_version = info.firmware_version
+
+            state = await self.hass.async_add_executor_job(self._device.status)
+            _LOGGER.debug("Got new state: %s", state)
+
+            self._carbon_dioxide_equivalent = state.co2e
+            self._relative_humidity = round(state.humidity, 1)
+            self._particulate_matter_2_5 = round(state.pm25, 1)
+            self._temperature = round(state.temperature, 1)
+            self._total_volatile_organic_compounds = round(state.tvoc, 3)
+
+        except DeviceException as ex:
+            _LOGGER.error("Got exception while fetching the state: %s", ex)
+
+    @property
+    def name(self):
+        """Return the name of this entity, if any."""
+        return self._name
+
+    @property
+    def device(self):
+        """Return the name of this entity, if any."""
+        return self._device
+
+    @property
+    def icon(self):
+        """Return the icon to use for device if any."""
+        return self._icon
+
+    @property
+    def manufacturer(self):
+        """Return the manufacturer version."""
+        return self._manufacturer
+
+    @property
+    def model(self):
+        """Return the device model."""
+        return self._model
+
+    @property
+    def sw_version(self):
+        """Return the software version."""
+        return self._sw_version
+
+    @property
+    def mac_address(self):
+        """Return the mac address."""
+        return self._mac_address
+
+    @property
+    def unique_id(self):
+        """Return the unique ID."""
+        return f"{self._model}-{self._mac_address}"
+
+    @property
+    def carbon_dioxide_equivalent(self):
+        """Return the CO2e (carbon dioxide equivalent) level."""
+        return self._carbon_dioxide_equivalent
+
+    @property
+    def relative_humidity(self):
+        """Return the humidity percentage."""
+        return self._relative_humidity
+
+    @property
+    def particulate_matter_2_5(self):
+        """Return the particulate matter 2.5 level."""
+        return self._particulate_matter_2_5
+
+    @property
+    def temperature(self):
+        """Return the temperature in °C."""
+        return self._temperature
+
+    @property
+    def total_volatile_organic_compounds(self):
+        """Return the total volatile organic compounds."""
+        return self._total_volatile_organic_compounds
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        data = {}
+
+        for prop, attr in PROP_TO_ATTR.items():
+            value = getattr(self, prop)
+            if value is not None:
+                data[attr] = value
+
+        return data
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def state(self):
```

Comment:
This should not be overridden. We already implement `particulate_matter_2_5` property. That's enough.